auto_submit/lib/configuration/repository_configuration_manager.dart

```diff
@@ -0,0 +1,174 @@
+  /// in the cache then go and get it from the repository and store it in the
+  /// cache.
+  ///
+  /// Entries will be stored in the cache as config/slug/autosubmit.yaml
+  Future<RepositoryConfiguration> readRepositoryConfiguration(
+    RepositorySlug slug,
+  ) async {
+    await _mutex.acquire();
+    try {
+      // Get the contents from the cache or go to github.
+      final cacheValue = await cache['${slug.fullName}$fileSeparator$fileName'].get(
+        () async => _getConfiguration(slug),
+        const Duration(minutes: 10),
+      );
+      // final String yamlContents = String.fromCharCodes(cacheValue!);
+      final String cacheYaml = String.fromCharCodes(cacheValue);
+      log.info('Converting yaml to RepositoryConfiguration: $cacheYaml');
+      return RepositoryConfiguration.fromYaml(cacheYaml);
+    } finally {
+      _mutex.release();
+    }
+  }
+
+  /// Collect the configuration from github and handle the cache conversion to
+  /// bytes.
+  Future<List<int>> _getConfiguration(
+    RepositorySlug slug,
+  ) async {
+    // Read the org level configuraiton file first.
+    log.info('Getting org level configuration.');
+    // Get the Org level configuration.
+    final RepositorySlug orgSlug = RepositorySlug(slug.owner, orgRepository);
+    final GithubService githubService = await config.createGithubService(orgSlug);
+    final String orgLevelConfig = await githubService.getFileContents(orgSlug, '$dirName$fileSeparator$fileName');
+    final RepositoryConfiguration globalRepositoryConfiguration = RepositoryConfiguration.fromYaml(orgLevelConfig);
+
+    // Collect the default branch if it was not supplied.
+    if (globalRepositoryConfiguration.defaultBranch!.isEmpty) {
+      globalRepositoryConfiguration.defaultBranch = await githubService.getDefaultBranch(slug);
+    }
+    log.info('Default branch was found to be ${globalRepositoryConfiguration.defaultBranch} for ${slug.fullName}.');
+
+    // If the override flag is set to true we check the pull request's
+    // repository to collect any values that will override the global config.
+    if (globalRepositoryConfiguration.allowConfigOverride == true) {
+      log.info('Override is set, collecting and merging local repository configuration.');
+      final GithubService localGithubService = await config.createGithubService(slug);
+
+      String? localRepositoryConfigurationYaml;
+      try {
+        localRepositoryConfigurationYaml =
+            await localGithubService.getFileContents(slug, '$dirName$fileSeparator$fileName');
+        final RepositoryConfiguration localRepositoryConfiguration =
+            RepositoryConfiguration.fromYaml(localRepositoryConfigurationYaml);
+        final RepositoryConfiguration mergedRepositoryConfiguration = mergeConfigurations(
+          globalRepositoryConfiguration,
+          localRepositoryConfiguration,
+        );
+        return mergedRepositoryConfiguration.toString().codeUnits;
+      } on Exception {
```

Comment:
Is there a more specific exception to track?

Reply:
Also should this log the exception and fail rather than hiding the error?

Reply:
Ah yeah for this we can fail. This was mostly due to testing as we had it configured and dart did not so I just left it as a warning knowing that we do not have overrides happening. If it is allowed and we find nothing would it be harmful to just use the global configuration?